### PR TITLE
[Serve] add dependencies on openssl

### PR DESCRIPTION
**Pull request by GeneDer**
## Why are these changes needed?

Add `pyOpenSSL` dependency for Serve. And update test docker file to use ray[serve-grpc] dependencies.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/requirements.txt [hunk 1]
```diff
@@ -57,3 +57,4 @@ pandas>=1.3
 pydantic!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,<3  # Serve users can use pydantic<2
 py-spy>=0.2.0
 memray; sys_platform != "win32" # memray is not supported on Windows
+pyOpenSSL
```
Comment (aslonnie) [inline python/requirements.txt:60]: please update the compiled dependency file?

Comment (GeneDer) [reply python/requirements.txt:60]: I think it already exist here https://github.com/ray-project/ray/blob/a31eede1155041eaec246b22c5362cac9fa8768d/python/requirements_compiled.txt#L1684 :)

Comment (aslonnie) [reply python/requirements.txt:60]: it still needs to be updated to include the line like:

```
-r /ray/ci/../python/requirements.txt
```

Comment (GeneDer) [reply python/requirements.txt:60]: I took the file from https://buildkite.com/ray-project/premerge/builds/29486#01922006-34c9-4366-a8b4-fd1a372567b2 and seems like that's the only change (besides added a new line in the end)

**Review by aslonnie — APPROVED** [review]

File: ci/docker/min.build.Dockerfile [hunk 2]
```diff
@@ -36,14 +36,11 @@ elif [[ "${EXTRA_DEPENDENCY}" == "ml" ]]; then
 elif [[ "${EXTRA_DEPENDENCY}" == "default" ]]; then
   pip-compile -o min_requirements.txt python/setup.py --extra default
 elif [[ "${EXTRA_DEPENDENCY}" == "serve" ]]; then
-  pip-compile -o min_requirements.txt python/setup.py --extra serve
+  pip-compile -o min_requirements.txt python/setup.py --extra serve-grpc
```
Comment (edoakes) [inline ci/docker/min.build.Dockerfile:39]: is serve-grpc a superset of serve ??

**Review by edoakes — APPROVED** [review]

File: ci/docker/min.build.Dockerfile (hunk 2 above)
Comment (edoakes) [reply ci/docker/min.build.Dockerfile:39]: answered my own question below

Comment (GeneDer) [reply ci/docker/min.build.Dockerfile:39]: yes!

**Pull request merged**